src/metax_api/services/file_service.py

```diff
@@ -734,19 +748,23 @@ def _get_directory_contents(cls, directory_id, recursive=False, max_depth=1, dep
                     cr_id,
                     dirs_only=dirs_only,
                     directory_fields=directory_fields,
-                    file_fields=file_fields)
+                    file_fields=file_fields,
+                    file_name=file_name,
+                    directory_name=directory_name
+                )
             except Http404:
                 if recursive:
                     return {'directories': []}
                 raise
         else:
             # browsing from ALL files, not cr specific
-            dirs = Directory.objects.filter(parent_directory_id=directory_id).values(*directory_fields)
-
+            dirs = Directory.objects.filter(parent_directory_id=directory_id, directory_name__icontains=directory_name)\
+                .values(*directory_fields)
```

Comment:
Had to test how ``directory_name__icontains=''`` is generated into SQL, and this is the result:

`AND UPPER("metax_api_directory"."directory_name"::text) LIKE UPPER('%%'))`

So it looks like if ``directory_name`` is an empty string, the query will have a slight performance cost in every request due to the last two procedure calls. This could be rather easily avoided with something like the following:

    dirs = Directory.objects.filter(parent_directory_id=directory_id).values(*directory_fields)

    if directory_name:
        dirs = dirs.filter(directory_name__icontains=directory_name)

As a side note, the above does not yet execute the query, so even more filters can be added later before executing if necessary.